src-electron/upgrade/upgrade.js

```diff
@@ -40,14 +48,48 @@ async function getForcedExternalStorage(db, attributeId) {
   let pkgs = await queryPackage.getPackageRefByAttributeId(db, attributeId)
   let zcl = await queryPackage.getPackageByPackageId(db, pkgs)
   zcl = zcl.path
-  let obj = await fsp.readFile(zcl)
-  let data = JSON.parse(obj)
+  let obj = await fsp.readFile(zcl, 'utf-8')
+  let data
+  if (isJsonString(obj)) {
+    data = JSON.parse(obj)
```

Comment:
So we are parsing as JSON twice?  Why?  Why not just have `maybeParseJsonString` or something that returns the parsed object or `undefined` if parsing throws?

Reply:
At one point this saved me from Buffer errors but those were solved by resolving promises properly. Will remove, thanks.